artemis/reporting/base/report.py

```diff
@@ -0,0 +1,112 @@
+import copy
+import datetime
+import socket
+import urllib.parse
+import uuid
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
+
+from artemis.reporting.utils import cached_gethostbyname
+from artemis.utils import get_host_from_url, is_ip_address
+
+from .normal_form import NormalForm
+from .report_type import ReportType
+from .reporters import get_all_reporters
+
+
+@dataclass
+class Report:
+    # top_level_target is the target that was provided when adding targets to be scanned. It may not be the same as
+    # the target where actual vulnerability was found - e.g. you may start with scanning example.com and the
+    # vulnerability may be found on https://subdomain.example.com/phpmyadmin/
+    top_level_target: str
+
+    # The actual location of the vulnerability (it may be e.g. a domain or a URL)
+    target: str
+    report_type: ReportType
+
+    # Additional report data - the content depends on the report type
+    report_data: Dict[str, Any]
```

Comment:
this field is already in `report` class, no need to have `report` in field name .